hs_app_timeseries/templates/pages/timeseriesresource.html

```diff
@@ -217,6 +244,303 @@ <h3 style="color: red" id="file-type-error"></h3>
+                    $('#id_longitude').attr('readonly', false);
+                    $("#id_site_type").css('pointer-events', 'auto');
+                    $("#id_elevation_datum").css('pointer-events', 'auto');
+                    var form_action = $('#id-site').attr('action');
+                    if(form_action.includes("update-metadata")) {
+                        var slice_end_index = form_action.toLowerCase().indexOf("site") + 5;
+                        form_action = form_action.slice(0, slice_end_index) + "add-metadata/";
+                        $('#id-site').attr('action', form_action);
+                    }
+                    return;
+                }
+
+                $(sites_json).each(function (i, site) {
+                    if (site.site_code === selectedSiteCode){
+                        $('#id_site_code').val(site.site_code);
+                        $('#id_site_name').val(site.site_name);
+                        $('#id_elevation_m').val(site.elevation_m);
+                        $('#id_elevation_datum').val(site.elevation_datum);
+                        $('#id_site_type').val(site.site_type);
+                        $('#id_latitude').val(site.latitude);
+                        $('#id_longitude').val(site.longitude);
+                        // change the site form action
+                        var form_action = $('#id-site').attr('action');
+                        if(form_action.includes("add-metadata")){
+                            form_action = form_action.replace("add-metadata", site.id + "/update-metadata");
+                        }
+                        else {
+                            var slice_end_index = form_action.toLowerCase().indexOf("site") + 5;
+                            form_action = form_action.slice(0, slice_end_index) + site.id + "/update-metadata/";
+                        }
+
+                        $('#id-site').attr('action', form_action);
+
+                        if (selectedSiteCode !== original_site_code){
+                            $('#id-site').find("button.btn-primary").show();
+                        }
+                        else {
+                            $('#id-site').find("button.btn-primary").hide();
+                        }
+                         $('#id_site_code').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_site_name').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_elevation_m').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_elevation_datum').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_site_type').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_latitude').attr('readonly', selectedSiteCode !== original_site_code);
+                         $('#id_longitude').attr('readonly', selectedSiteCode !== original_site_code);
```

Comment:
@pkdash I think a for loop with array would work better to reduce code duplication, especially for extensibility considering in the future you may want to update more attributes or change css styles to all these elements together.  This is a minor point, but I think it has value if you have more than 5 elements to update in the same way.